### DISCUSSION TITLE
How-To Update from archive with composer

### DISCUSSION THREAD

**Discussion by alexh-swdev** [OpenMage/magento-lts#4094] [Questions & Answers]
I have a seperate repo where i always push the OM releases to. Therefore, I am using usually the full upgrade zips (the one with the vendor folder etc.) from the release and merge it in.
(Then I diff the changes between the releases and apply it to the custom shops)

Because of https://github.com/OpenMage/magento-lts/issues/4093 I tried using the src archive (without the vendor etc. folder) and run `composer install --no-dev --optimize-autoloader -n` from the root folder. In the end I am getting errors about failiures when applying patches to zf1-future:
```
 - Applying patches for shardj/zf1-future
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/MAG-1.1.1.patch (MAG-1.1.1)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/MAG-1.1.1.patch
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/MAG-1.9.3.0.patch (MAG-1.9.3.0)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/MAG-1.9.3.0.patch
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/MAG-1.9.3.7.patch (MAG-1.9.3.7 - SUPEE-10415)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/MAG-1.9.3.7.patch
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-918.patch (OM-918 - Add runtime cache to Zend_Locale_Data)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-918.patch
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-1081.patch (OM-1081 - Not detecting HTTPS behind a proxy)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-1081.patch
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-2047.patch (OM-2047 - Pass delimiter char to preg_quote)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-2047.patch
    https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-2050.patch (OM-2050 - Prevent checking known date codes)
   Could not apply patch! Skipping. The error was: Cannot apply patch https://raw.githubusercontent.com/OpenMage/composer-patches/74862e7a9fa29dd62c5c02d585d931ce9e3c5815/patches/OM-2050.patch

Package eloquent/enumeration is abandoned, you should avoid using it. No replacement was suggested.
```

Can I ignore those? (I guess, not :) )
What did I do wrong?

**Comment by luigifab** [comment]
I'm not sure how the zip with the vendor directory is generated, but I suggest you to move your installation to composer. It's time consuming, but it's the old future!

**Reply by alexh-swdev** [reply]
Yes, that's what I did: I downlaoded the src.zip (the one w/o the vendor) and ran the composer install like mentioned above. But it fails when applying the patches. As I didn't find a doc about upgrading via the src.zip, I don't know whether I am doing sth wrong or the composer script is faulty?

**Reply by luigifab** [reply]
oh yes no that's not what I mean
you can't use the zip without the vendor directory

I suggested to you to install OpenMage with composer

**Reply by alexh-swdev** [reply]
So I did a fresh composer install but still, the problem persists. Patches cannot get applied :(
```
composer --version
Composer version 2.7.7 2024-06-10 22:11:12
PHP version 8.3.2 (C:\PHP\8.3\php.exe)
```

The current resulting OM composer.json:
```
{
    "name": "swh/openmagerelease",
    "require": {
        "aydin-hassan/magento-core-composer-installer": "2.1.0",
        "openmage/magento-lts": "20.10.0"
    },
    "extra": {
        "enable-patching": true,
        "magento-core-package-type": "magento-source",
        "magento-root-dir": "."
    },
    "config": {
        "allow-plugins": {
            "aydin-hassan/magento-core-composer-installer": true,
            "cweagans/composer-patches": true,
            "magento-hackathon/magento-composer-installer": true
        }
    }
}

```

Probably time to turn this into an issue?
Or isn't this supposed to run on a Windows system?
Are there logfiles with more infos? (Usually, if I have to use composer, it "just works" :p )

**Reply by sreichel** [reply]
Can you please add verbose output? (--v)

**Reply by addison74** [reply]
I haven't encountered this problem so far, but it can be quickly verified in a DDEV test environment. I doubt that it is from the version of Composer used.

I have never used the zip version, only the tar.gz version because I prefer to let Composer create the vendor directory. I also installed OpenMage with Composer and cloned the repository, and the patches were applied.

I will test again and let you know the we decide if it should be reported as an issue.

**Reply by alexh-swdev** [reply]
Addison, I threw away the zip version and did a plain composer install this time. (just to clarify 'cause of "never used the zip version" :) )
Sven, I'll get back with --v

**Reply by addison74** [reply]
Composer installation should be fine. I did it last week when I added a few custom patches for ZF1-Future.

**Reply by alexh-swdev** [reply]
`composer install --verbose` revealed "patch not found" 
=> Found a version of patch in the git install, so I added `C:\Program Files\Git\usr\bin\` to the user path var and now it works.

Thanks for putting me in the right direction!

**Reply by sreichel** [reply]
Better use ddev + wsl2  ... could make your life easier :)

**Reply by addison74** [reply]
```
ddev ssh
composer init + all steps from documentation
composer require "openmage/magento-lts":"^20.0.10"
```

![screenshot](https://github.com/user-attachments/assets/64548031-b1c0-4102-808f-6b6604be865c)

**Reply by addison74** [reply]
> Better use ddev + wsl2 ... could make your life easier :)

Definitely the problem is coming from his test environment.

I recommend DDEV too https://github.com/OpenMage/magento-lts/discussions/3839.

**Reply by sreichel** [reply]
Easiest ... run Linux via VirtualBox and forget about all that Windows workarounds 😀

**Reply by alexh-swdev** [reply]
Oh boys, you make my stomach hurt... (such things in a minor release...) 
![grafik](https://github.com/user-attachments/assets/7bbca05d-436c-4ad2-a7f5-03af08bf0721)

And yes, I am already using WSL2 ;) But getting the releases and syncing them to my development repos is (was?) done on the Windows side until now. Does the PHPStorm auto-sync-on-save-via-ssh-into-wsl2 keep those symlinks or does it remove them?

**Reply by alexh-swdev** [reply]
Sven, Don't laugh. I just really thought about the VBox. (Or hyper-v - if it can handle linux) :P
Ah sometimes I really hate windows ;)

**Reply by sreichel** [reply]
I still run Win7 ... VBox is easy to setup, it's nearly fast as native ... and no fu** with wsl configuration ...  :)

**Reply by addison74** [reply]
I didn't notice any differences in performance between (Windows 10 + WSL2 + Docker) and (Linux + Docker), but in Windows I need at least 16 MB of RAM to run decently with PHPStorm open. I work in a VMware where I created my development/testing environment.
